File: app/src/main/java/com/example/dulong/activity/product/ProductDetailActivity.java
package com.example.dulong.activity.product;

import android.os.Bundle;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.bumptech.glide.Glide;
import com.example.dulong.R;
import com.example.dulong.activity.utils.CartManager;
import com.example.dulong.model.Product;
import com.google.android.material.appbar.MaterialToolbar;

import java.text.DecimalFormat;

public class ProductDetailActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_product_detail);

        // 1. Ánh xạ View
        MaterialToolbar toolbar = findViewById(R.id.toolbar);
        ImageView ivImage = findViewById(R.id.iv_product_image);
        TextView tvName = findViewById(R.id.tv_product_name);
        TextView tvPrice = findViewById(R.id.tv_product_price);
        TextView tvDescription = findViewById(R.id.tv_product_description);
        Button btnAddToCart = findViewById(R.id.btn_add_to_cart);

        // Thông số kỹ thuật
        TextView tvWeight = findViewById(R.id.tv_product_weight);
        TextView tvBalance = findViewById(R.id.tv_product_balance);
        TextView tvFlex = findViewById(R.id.tv_product_flex);
        TextView tvColor = findViewById(R.id.tv_product_color);
        TextView tvQuantity = findViewById(R.id.tv_product_quantity);

        toolbar.setNavigationOnClickListener(v -> finish());

        // 2. Nhận dữ liệu
        Product product = (Product) getIntent().getSerializableExtra("PRODUCT_DATA");

        if (product != null) {
            tvName.setText(product.getName());
            DecimalFormat formatter = new DecimalFormat("#,###");
            tvPrice.setText(formatter.format(product.getPrice()) + "đ");
            tvDescription.setText(product.getDescription() != null ? product.getDescription() : "Chưa có mô tả.");

            // Hiển thị thông số
            tvWeight.setText("Trọng lượng: " + (product.getWeight() != null ? product.getWeight() : "..."));
            tvBalance.setText("Điểm cân bằng: " + (product.getBalance() != null ? product.getBalance() : "..."));
            tvFlex.setText("Độ cứng: " + (product.getFlex() != null ? product.getFlex() : "..."));
            tvColor.setText("Màu sắc: " + (product.getColor() != null ? product.getColor() : "..."));

            // Kiểm tra số lượng tồn kho (giả sử > 0 là còn hàng)
            // Lưu ý: Trong CartManager ta dùng 'quantity' là số lượng mua
            // Còn ở đây product.quantity có thể hiểu là tồn kho nếu API trả về
            if (product.getQuantity() >= 0) {
                tvQuantity.setText("Tình trạng: Còn hàng");
                btnAddToCart.setEnabled(true);
            } else {
                tvQuantity.setText("Hết hàng");
                btnAddToCart.setEnabled(false);
                btnAddToCart.setText("Hết hàng");
            }

            Glide.with(this)
                    .load(product.getImage())
                    .placeholder(R.drawable.ic_logo_only)
                    .into(ivImage);

            // 3. Xử lý thêm vào giỏ
            btnAddToCart.setOnClickListener(v -> addToCart(product));
        } else {
            Toast.makeText(this, "Lỗi dữ liệu!", Toast.LENGTH_SHORT).show();
            finish();
        }
    }

    private void addToCart(Product product) {
        CartManager.addToCart(product);
        Toast.makeText(this, "Đã thêm vào giỏ hàng!", Toast.LENGTH_SHORT).show();
    }
}

File: app/src/main/java/com/example/dulong/activity/product/ProductDetailActivity.kt
package com.example.dulong.activity.product

import android.annotation.SuppressLint
import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.bumptech.glide.Glide
import com.example.dulong.activity.cart.CardActivity // Hoặc CardActivity tùy tên file của bạn
import com.example.dulong.activity.utils.CartManager
import com.example.dulong.R
import com.example.dulong.model.Product
import com.google.android.material.appbar.MaterialToolbar
import java.text.DecimalFormat

class ProductDetailActivity : AppCompatActivity() {

    @SuppressLint("SetTextI18n")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_product_detail)

        // 1. Ánh xạ View
        val toolbar: MaterialToolbar = findViewById(R.id.toolbar)
        val ivImage: ImageView = findViewById(R.id.iv_product_image)
        val tvName: TextView = findViewById(R.id.tv_product_name)
        val tvPrice: TextView = findViewById(R.id.tv_product_price)
        val tvDescription: TextView = findViewById(R.id.tv_product_description)
        val btnAddToCart: Button = findViewById(R.id.btn_add_to_cart)

        // Thông số kỹ thuật
        val tvWeight: TextView = findViewById(R.id.tv_product_weight)
        val tvBalance: TextView = findViewById(R.id.tv_product_balance)
        val tvFlex: TextView = findViewById(R.id.tv_product_flex)
        val tvColor: TextView = findViewById(R.id.tv_product_color)
        val tvQuantity: TextView = findViewById(R.id.tv_product_quantity)

        toolbar.setNavigationOnClickListener { finish() }

        // 2. Nhận dữ liệu
        val product = intent.getSerializableExtra("PRODUCT_DATA") as? Product

        if (product != null) {
            tvName.text = product.name
            val formatter = DecimalFormat("#,###")
            tvPrice.text = "${formatter.format(product.price)}đ"
            tvDescription.text = product.description ?: "Chưa có mô tả."

            // Hiển thị thông số
            tvWeight.text = "Trọng lượng: ${product.weight ?: "..."}"
            tvBalance.text = "Điểm cân bằng: ${product.balance ?: "..."}"
            tvFlex.text = "Độ cứng: ${product.flex ?: "..."}"
            tvColor.text = "Màu sắc: ${product.color ?: "..."}"

            // Kiểm tra số lượng tồn kho (giả sử > 0 là còn hàng)
            // Lưu ý: Trong CartManager ta dùng 'quantity' là số lượng mua
            // Còn ở đây product.quantity có thể hiểu là tồn kho nếu API trả về
            if (product.quantity >= 0) {
                tvQuantity.text = "Tình trạng: Còn hàng"
                btnAddToCart.isEnabled = true
            } else {
                tvQuantity.text = "Hết hàng"
                btnAddToCart.isEnabled = false
                btnAddToCart.text = "Hết hàng"
            }

            Glide.with(this)
                .load(product.image)
                .placeholder(R.drawable.ic_logo_only)
                .into(ivImage)

            // 3. Xử lý thêm vào giỏ
            btnAddToCart.setOnClickListener {
                addToCart(product)
            }
        } else {
            Toast.makeText(this, "Lỗi dữ liệu!", Toast.LENGTH_SHORT).show()
            finish()
        }
    }

    private fun addToCart(product: Product) {
        CartManager.addToCart(product)
        Toast.makeText(this, "Đã thêm vào giỏ hàng!", Toast.LENGTH_SHORT).show()
    }
}